core/trino-main/src/test/java/io/trino/execution/TestAddColumnTask.java

```diff
@@ -0,0 +1,137 @@
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", tableName);
+    }
+
+    @Test
+    public void testAddColumnNotExistingTableIfExists()
+    {
+        QualifiedName tableName = qualifiedName("not_existing_table");
+
+        getFutureValue(executeAddColumn(tableName, new Identifier("test"), INTEGER, Optional.empty(), true, false));
+        // no exception
+    }
+
+    @Test
+    public void testAddColumnNotExists()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+
+        getFutureValue(executeAddColumn(asQualifiedName(tableName), new Identifier("test"), INTEGER, Optional.empty(), false, true));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("test", BIGINT)));
+    }
+
+    @Test
+    public void testAddColumnOnView()
+    {
+        QualifiedObjectName viewName = qualifiedObjectName("existing_view");
+        metadata.createView(testSession, viewName, someView(), false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(viewName), new Identifier("test"), INTEGER, Optional.empty(), false, false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", viewName);
+    }
+
+    @Test
+    public void testAddColumnOnMaterializedView()
+    {
+        QualifiedObjectName materializedViewName = qualifiedObjectName("existing_materialized_view");
+        metadata.createMaterializedView(testSession, QualifiedObjectName.valueOf(materializedViewName.toString()), someMaterializedView(), false, false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(materializedViewName), new Identifier("test"), INTEGER, Optional.empty(), false, false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", materializedViewName);
+    }
+
+    private ListenableFuture<Void> executeAddColumn(QualifiedName table, Identifier column, Type type, Optional<String> comment, boolean tableExists, boolean columnNotExists)
+    {
+        ColumnDefinition columnDefinition = new ColumnDefinition(column, toSqlType(type), true, ImmutableList.of(), comment);
+        return new AddColumnTask(plannerContext, new AllowAllAccessControl(), columnPropertyManager)
+                .execute(new AddColumn(new NodeLocation(1, 1), table, columnDefinition, tableExists, columnNotExists), queryStateMachine, ImmutableList.of(), WarningCollector.NOOP);
```

Comment:
Any specific reason for this `NodeLocation` can it be `Optional#empty` or invoke the other ctor which doesn't involve `NodeLocation`

Reply:
Ditto for `TestDropColumnTask`